frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -1184,7 +1126,38 @@ export default ({
         e?.dataTransfer.files?.length &&
           this.$refs.sendArea.fileAttachmentHandler(e?.dataTransfer.files, true)
       }
-    }
+    },
+    processSwitchQueue: debounce(async function () {
+      if (this.ephemeral.chatroomSwitchQueue.length === 0) return
+
+      // Take the most recent chatroom entry on the queue and discard everything else. This way we can speed up the process of switching chatrooms.
+      const targetChatroomId = this.ephemeral.chatroomSwitchQueue.pop()
+      this.ephemeral.chatroomSwitchQueue = []
```

Comment:
I'm not sure why an array is needed here, when the array-property isn't really used.

Reply:
Its theoretically possible for there to be multiple elements in the list if the chatroom switched really quickly